filesmanager/tests/test_filesmanager.py

```diff
@@ -0,0 +1,949 @@
+        Set up the test suite.
+        """
+        self.xblock = FilesManagerXBlock(
+            runtime=Mock(),
+            field_data=Mock(),
+            scope_ids=Mock(),
+        )
+        self.xblock.fill_unpublished = Mock()
+        self.xblock.get_current_user = Mock()
+        self.xblock.initialize_directories = Mock()
+        self.xblock.temporary_save_upload_files = Mock()
+        self.xblock._sync_content = Mock()  # pylint: disable=protected-access
+        self.xblock.clean_uploaded_files = Mock()
+        self.xblock.delete_asset = Mock()
+        self.xblock.content_paths = []
+        self.xblock.temporary_uploaded_files = {}
+        self.xblock.directories = {
+            "id": "root",
+            "name": "Root",
+            "type": "directory",
+            "path": "Root",
+            "parentId": "",
+            "metadata": {},
+            "children": [
+                {
+                    "id": "unpublished",
+                    "parentId": "root",
+                    "name": "Unpublished",
+                    "type": "directory",
+                    "path": "Root/Unpublished",
+                    "metadata": {},
+                    "children": [],
+                }
+            ],
+        }
+
+
+class TestFilesManagerXBlockHandlers(FilesManagerXBlockTestMixin):
+    """
+    Test suite for the FilesManagerXBlock JSON handlers.
+    """
+
+    def setUp(self) -> None:
+        """
+        Set up the test suite.
+        """
+        super().setUp()
+        self.request = Mock(
+            body=json.dumps({}).encode("utf-8"),
+            method="POST",
+            status_code_success=HTTPStatus.OK,
+        )
+
+    def test_get_directories_anonymous_user(self):
+        """
+        Check get directories JSON handler for anonymous user.
+
+        Expected result:
+            - The view returns 200 status code.
+            - The directories list are returned without the unpublished directory.
+        """
+        self.xblock.get_current_user.return_value.opt_attrs.get.return_value = True
+        expected_result = {
+            "status": "success",
+            "contents": {
+                "id": "root",
+                "name": "Root",
+                "type": "directory",
+                "path": "Root",
+                "parentId": "",
+                "metadata": {},
+                "children": [],
+            },
+        }
+
+        response = self.xblock.get_directories(self.request)
+
+        self.xblock.fill_unpublished.assert_not_called()
+        self.assertEqual(HTTPStatus.OK, response.status_code)
+        self.assertEqual(expected_result, response.json)  # pylint: disable=no-member
+
+    def test_get_directories_non_anonymous_user(self):
+        """
+        Check get directories JSON handler for non-anonymous user.
+
+        Expected result:
+            - The view returns 200 status code.
+            - The directories list are returned with the unpublished directory.
```

Comment:
This should happen if and only if:
- The user is not a student
- We're in the studio edit view
The first check I believe is happening in line 110, is the other one as well?

Reply:
Yes, the only way for `self.get_current_user().opt_attrs.get(ATTR_KEY_ANONYMOUS_USER_ID)` to return `None` is for these two conditions to be met